model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/stencils/calculate_nabla2_and_smag_coefficients_for_vn.py

```diff
@@ -39,111 +39,36 @@ def _calculate_nabla2_and_smag_coefficients_for_vn(
         (diff_multfac_smag, u_vert, v_vert, smag_offset), wpfloat
     )
 
-    dvt_tang_wp = (
-        -(
-            u_vert_wp(E2C2V[0]) * dual_normal_vert_x[E2C2VDim(0)]
-            + v_vert_wp(E2C2V[0]) * dual_normal_vert_y[E2C2VDim(0)]
-        )
-    ) + (
-        u_vert_wp(E2C2V[1]) * dual_normal_vert_x[E2C2VDim(1)]
-        + v_vert_wp(E2C2V[1]) * dual_normal_vert_y[E2C2VDim(1)]
-    )
+    v_n = u_vert_wp(E2C2V) * primal_normal_vert_x + v_vert_wp(E2C2V) * primal_normal_vert_y
 
-    dvt_norm_vp = astype(
-        (
-            -(
-                u_vert_wp(E2C2V[2]) * dual_normal_vert_x[E2C2VDim(2)]
-                + v_vert_wp(E2C2V[2]) * dual_normal_vert_y[E2C2VDim(2)]
-            )
-        )
-        + (
-            u_vert_wp(E2C2V[3]) * dual_normal_vert_x[E2C2VDim(3)]
-            + v_vert_wp(E2C2V[3]) * dual_normal_vert_y[E2C2VDim(3)]
-        ),
-        vpfloat,
+    nabla2_of_vn = (v_n[E2C2VDim(0)] + v_n[E2C2VDim(1)] - wpfloat("2.0") * vn) * (
+        inv_primal_edge_length * inv_primal_edge_length
```

Comment:
Here we could also use `l_p*l_p` from below, however this might not be bit identical as `l_p` contains a multiplication with +/-1 (which drops out in the square)